pkg/controller/dataimportcron-controller.go

```diff
@@ -808,20 +822,25 @@ func (r *DataImportCronReconciler) newCronJob(cron *cdiv1.DataImportCron) (*batc
 		)
 	}
 
+	addEnvVar := func(varName, value string) {
+		container.Env = append(container.Env, corev1.EnvVar{Name: varName, Value: value})
+	}
+
 	if insecureTLS {
-		container.Env = append(container.Env,
-			corev1.EnvVar{
-				Name:  common.InsecureTLSVar,
-				Value: "true",
-			},
-		)
+		addEnvVar(common.InsecureTLSVar, "true")
 	}
 
-	successfulJobsHistoryLimit := int32(0)
-	failedJobsHistoryLimit := int32(0)
-	ttlSecondsAfterFinished := int32(0)
-	backoffLimit := int32(2)
-	gracePeriodSeconds := int64(0)
+	addEnvVarFromImportProxyConfig := func(varName string) {
+		if value, err := GetImportProxyConfig(cdiConfig, varName); err == nil {
+			addEnvVar(varName, value)
+		} else {
+			r.log.Info("Missing", varName, err.Error())
+		}
+	}
+
+	addEnvVarFromImportProxyConfig(common.ImportProxyHTTP)
```

Comment:
So adding the env vars is enough to ensure the proxy is respected?
@awels you identified some golang specifics with regards to HTTP env vars before, will this be fine?

Reply:
No, the CronJob poller app was adapted to use the proxy certs. Maybe these env vars are just leftovers of my debug and not needed at all. Let me check...

Reply:
I am not sure how we will be proxying stuff without the env vars, isn't that the only magic that makes proxying take place? how else would the go/docker client know to proxy it?

Reply:
We pass the proxy certs to the pod and the containerized app (poller) use it, 
so it can poll registry url via the configured proxy. We don't initiate any proxying here.

Reply:
Without the proxy certs the poller app failed on connecting to the proxy. This was solved.

Reply:
I guess it proxied after the first import or something @awels ?

Reply:
No the proxying is enabled when the environment variables are set `http_proxy, https_proxy, no_proxy` some applications want to use upper case version of that. The golang client uses the lower case versions though.

The extra certs are basically needed to connect to the proxy if the proxy has self signed CAs.